File: practice/algorithms/greedy/max_min/MaxMin.js
'use strict';

const processData = input => {
    let lines = input.split('\n').map(i => parseInt(i));
    let k = lines[1];
    let sortedLines = lines.slice(2).sort((l,j) => l - j);
    let unfairness = sortedLines[sortedLines.length - 1];
    for(let i = 0; i <= sortedLines.length - k; i++) {
        let temp = sortedLines[i + k - 1] - sortedLines[i];
        if(unfairness > temp) {
            unfairness = temp
        }
    }
    console.log(unfairness);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/greedy/max_min/MaxMin.java
import java.util.*;

public class MaxMin {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = Integer.parseInt(stdin.nextLine());
        int k = Integer.parseInt(stdin.nextLine());
        int[] candies = new int[tests];
        for(int i = 0; i < tests; i++) {
            candies[i] = Integer.parseInt(stdin.nextLine());
        }
        Arrays.sort(candies);
        int unfairness = candies[candies.length - 1];
        for(int i = 0; i <= candies.length - k; i++) {
            int temp = candies[i + k - 1] - candies[i];
            if(unfairness > temp) {
                unfairness = temp;
            }
        }
        System.out.println(unfairness);
        stdin.close();
    }
}